files/common/usr/bin/unpack-image

```diff
@@ -96,22 +117,35 @@ tar -xzf "$UPGRADE_IMAGE_PATH" ||
 
 report_progress_inc 40 "Verifying format."
 
+# check if this is new style upgrade image
+if [[ -f "complete-unpack" ]]; then
+
+	# verify signatures and hashes before hand-off
+	verify_sig $opt_s
```

Comment:
Can we restructure this better, so that both code paths (with and without "complete-unpack") can call this logic at the same place (in which case, a function probably won't been required)? To me, having both code paths do the same thing, but in two different places in the script, makes things more difficult to read and understand (even if the actual logic is in a function)..

Reply:
yes I agree. I will move this above the if block.